test/geometry_tests.cpp

```diff
@@ -61,105 +94,176 @@ bool checkFirstOrLastPoint(int firstOrLastPoint, int x_shouldbe, int y_shouldbe,
 
 }
 
+std::ostream& operator<<(std::ostream& os, const std::vector<Cell>& v){
+	os << "\n";
+	for (const auto& pixel:v) {
+		os << "(" << pixel.x << "," << pixel.y << ")\n";
+	}
+	os << "\n";
+	return os;
+}
+
+inline bool operator==(const Cell &lhs,const Cell &rhs) {
+	return (lhs.x == rhs.x) && (lhs.y == rhs.y);
+}
+
 TEST(Point2DPlanning_Griz_Test, bresenham_conversion_test) {
 	// Get the points for the line connecting (0,0) and (5,0)
-	auto v = bresenham_conversion(0, 0, 5, 0);
+	const auto v = bresenham_conversion(0, 0, 5, 0);
 
 	// Check if the vector size is 6 points 
-	ASSERT_EQ(v.size(), 6);
+	EXPECT_THAT(v.size(), testing::Eq(6));
 
 	// Check if the first point is (0,0)
-	ASSERT_EQ(checkFirstOrLastPoint(0, 0, 0, v), true);
+	EXPECT_THAT(v.front(), testing::ElementsAre(0,0));
 
 	// Check if the last point is (5,0)
-	ASSERT_EQ(checkFirstOrLastPoint(1, 5, 0, v), true);
+	// EXPECT_THAT(checkFirstOrLastPoint(1, 5, 0, v), testing::IsTrue());
+	EXPECT_THAT(v.back(), testing::ElementsAre(5,0));
 }
 
 TEST(Point2DPlanning_Griz_Test, bresenham_conversion_horz_line_test) {
 	// Create a horizontal line between points (0,0) and (5,0)
-	auto v = bresenham_conversion(0, 0, 5, 0);
+	const auto v1 = bresenham_conversion(0, 0, 5, 0);
 		
 	// Check there are 6 points
-	ASSERT_EQ(v.size(), 6);
+	EXPECT_THAT(v1.size(), testing::Eq(6));
 
 	// Check that all the y coordinates are the same (indicates horz line)
-	ASSERT_EQ(allXOrYCoorSame(1,v), true);	
+	EXPECT_THAT(allXOrYCoorSame(1,v1), testing::IsTrue());	
 
 	// Create a horizontal line from (4,5) to (-1,5)
-	v = bresenham_conversion(4, 5, -1, 5);
+	const auto v2 = bresenham_conversion(4, 5, -1, 5);
```

Comment:
You shouldn't be doing multiple bresenham in one test afaik

Reply:
Done